### PR TITLE
CI Fix

### LINKED ISSUE #991: integration tests are failing
https://github.com/ploomber/jupysql/actions/runs/7824421067/job/21346960913

### PR DESCRIPTION
**Pull request by neelasha23**
## Describe your changes

1. Removed test `test_db_driver_error_raised` because the latest DuckDB release is handling these cases:

Screenshot 2024-02-22 at 12 20 26 PM

2. modified the sqlglot parsing logic to check for Nonetype.

Opened issues for the pending fixes:
https://github.com/ploomber/jupysql/issues/994
https://github.com/ploomber/jupysql/issues/993

## Issue number

Closes #991 

## Checklist before requesting a review

- [ ] Performed a self-review of my code
- [ ] Formatted my code with [`pkgmt format`](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#linting-formatting)
- [ ] Added [tests](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#testing) (when necessary).
- [ ] Added [docstring](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#documenting-changes-and-new-features) documentation and update the [changelog](https://ploomber-contributing.readthedocs.io/en/latest/contributing/pr.html#changelog) (when needed)

----
📚 Documentation preview 📚: https://jupysql--992.org.readthedocs.build/en/992/

### CODE REVIEW

**Comment by neelasha23** [comment]
Please review @edublancas

File: src/sql/util.py [hunk 1]
```diff
@@ -530,7 +530,11 @@ def extract_tables_from_query(query):
         [] if error in parsing the query
     """
     try:
-        tables = [table.name for table in parse_one(query).find_all(exp.Table)]
+        tables = [
+            table.name
+            for table in parse_one(query).find_all(exp.Table)
+            if hasattr(table, "name")
+        ]
```
Comment (edublancas) [inline src/sql/util.py:537]: why is this change required?

File: src/tests/test_magic_plot.py [hunk 2]
```diff
@@ -469,11 +469,13 @@ def test_bar_two_col(load_data_two_col, ip):
 
 
 @_cleanup_cm()
+@pytest.mark.xfail(reason="Failing intermittently with DuckDB v0.10.0")
```
Comment (edublancas) [inline src/tests/test_magic_plot.py:472]: if duckdb is the problem, it'd be better to test with <0.10.0 and open an issue

please open the issue and then document why 0.10.0 is failing

**Review by edublancas — CHANGES_REQUESTED** [review]

File: src/sql/util.py (hunk 1 above)
Comment (neelasha23) [reply src/sql/util.py:537]: Previous version of sqlglot was returinng empty list if no tables found, currently it return None.

File: src/tests/test_magic_plot.py (hunk 2 above)
Comment (neelasha23) [reply src/tests/test_magic_plot.py:472]: Have tagged a couple of issues in the issue description, please check if these are ok

**Comment by edublancas** [comment]
@neelasha23 the tests are still failing: https://github.com/ploomber/jupysql/actions/runs/8007801460/job/21872791573?pr=992

**Comment by neelasha23** [comment]
> @neelasha23 the tests are still failing: https://github.com/ploomber/jupysql/actions/runs/8007801460/job/21872791573?pr=992

Yes it's the clickhouse tests only. Opened an issue and tagged in the PR description: https://github.com/ploomber/jupysql/issues/993

I'm still figuring out the issue @edublancas

**Review by edublancas — APPROVED** [review]
